core/src/main/scala/org/apache/spark/rdd/InputFileBlockHolder.scala

```diff
@@ -41,25 +43,26 @@ private[spark] object InputFileBlockHolder {
    * The thread variable for the name of the current file being read. This is used by
    * the InputFileName function in Spark SQL.
    */
-  private[this] val inputBlock: InheritableThreadLocal[FileBlock] =
-    new InheritableThreadLocal[FileBlock] {
-      override protected def initialValue(): FileBlock = new FileBlock
+  private[this] val inputBlock: InheritableThreadLocal[AtomicReference[FileBlock]] =
+    new InheritableThreadLocal[AtomicReference[FileBlock]] {
```

Comment:
Does it mean, parent and child threads share the same `FileBlock` object?

Reply:
More specifically, it means it holds one reference where `FileBlock` instance is atomically set and unset.